internal/provider/clickhouse_data_source.go

```diff
@@ -187,6 +224,40 @@ func parseClickhousePrivateConnectionInfo(r *clickhouse.PrivateConnectionInfo) *
 	return c
 }
 
+func parseClickhouseCustomCertificate(r *clickhouse.CustomCertificate, oldKey string, diags diag.Diagnostics) *ClickhouseCustomCertificate {
+	if r == nil {
+		return nil
+	}
+
+	if !r.GetEnabled() {
+		return nil
+	}
+
+	certRaw := string(r.Certificate.GetValue()[:])
+	if len(certRaw) == 0 {
+		return nil
+	}
+
+	certificate := types.StringValue(certRaw)
+	s, err := strconv.Unquote(oldKey)
```

Comment:
I suppose we need to be able to support multiple formats here, or describe which format of keys we support.

Reply:
Most of users will use `file` [1] function to forward keys, it's gonna be unquoted for most of cases.

[1] https://developer.hashicorp.com/terraform/language/functions/file

Reply:
Works as with
```
key = file("server.key")
```
as well as with
```
key = <<EOT
-----BEGIN CERTIFICATE-----
....
-----END CERTIFICATE-----
EOT
```